src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,193 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+    private Node<T> newNode;
+
+    public MyLinkedList() {
+        first = null;
+        last = null;
+    }
+
+    public Node<T> nodeByIndex(int index) {
+        if (index < size / 2) {
+            newNode = first;
+            for (int i = 0; i < index; i++) {
+                newNode = newNode.next;
+            }
+            return newNode;
+        } else {
+            newNode = last;
+            for (int i = size - 1; i > index; i--) {
+                newNode = newNode.prev;
+            }
+            return newNode;
+        }
+    }
+
+    public void addFirst(T value) {
+        newNode = new Node(value);
+        first = last = newNode;
+    }
+
+    public void addFront(T value) {
+        newNode = new Node(value);
+        first.prev = newNode;
+        newNode.next = first;
+        newNode.prev = null;
+        first = newNode;
+    }
+
+    public void addBack(T value) {
+        newNode = new Node(value);
+        newNode.prev = last;
+        last.next = newNode;
+        newNode.next = null;
+        last = newNode;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == 0) {
+            addFirst(value);
+        } else {
+            addBack(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size < index || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == 0) {
+            if (size == 0) {
+                addFirst(value);
+            } else {
+                addFront(value);
+            }
+        }
+        if (index > 0) {
+            if (index == size) {
+                addBack(value);
+            }
+            newNode = new Node(value);
+            nodeByIndex(index - 1).next = newNode;
```

Comment:
try doesn't call this method `nodeByIndex()` thrice

Reply:
ok